daemon/cmd/daemon.go

```diff
@@ -649,6 +651,16 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 				option.EnableHostLegacyRouting)
 		}
 	}
+	if option.Config.EnableEgressGateway {
+		kernelVersion, err := version.GetKernelVersion()
+		if err != nil {
+			return nil, nil, fmt.Errorf("cannot determine kernel version: %w", err)
+		}
+
+		if versioncheck.MustCompile("<" + "5.4.0")(kernelVersion) {
```

Comment:
Is there a way to perform this check without tying it directly to a kernel version? Typically RHEL backports a lot of functionality, and this can mean that Cilium refuses to run this logic on a RHEL kernel when it should really work there.

Ideally this should probe for the actual kernel feature this is relying on from v5.4.

Reply:
I think we can probably probe for `HAVE_LARGE_INSN_LIMIT` and use that instead of the kernel version